nixos/modules/services/networking/wireguard.nix

```diff
@@ -201,43 +230,30 @@ let
         };
 
         script = ''
-          modprobe wireguard
+	  ip link set up dev ${name}
 
-          ${values.preSetup}
-
-          ip link add dev ${name} type wireguard
-
-          ${concatMapStringsSep "\n" (ip:
-            "ip address add ${ip} dev ${name}"
-          ) values.ips}
-
-          wg set ${name} private-key ${privKey} ${
-            optionalString (values.listenPort != null) " listen-port ${toString values.listenPort}"}
+	  # wait for ${name} to be configured
+	  while ! networkctl status $DEVICE | grep configured 1>/dev/null; do
```

Comment:
Ugh, this is ugly. What if the users restart systemd-networkd manually?

Reply:
Could this be fixed with PartOf=systemd-networkd.service?

Also the indentation is off in a few places.

Reply:
Maybe?

Reply:
Also then module should set the device up with `ip` after networkctl has configured the device. Does not networkd also handles that?

Reply:
This could be fixed by waiting `systemd-network-wait-online@${name}.service`. How could we make this unit from `systemd-network-wait-online@.service`?

The while loop could be replaced with `${config.systemd.package}/lib/systemd/systemd-networkd-wait-online -i ${name}`. If systemd-networkd is restarted, the keys are restored as expected.